[staging] Build failure: python3.pkgs.beautifulsoup4

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix build nixpkgs/dc05cdb0ced63fee5124d22a5cc43a0eab3a23bb#python3.pkgs.beautifulsoup4`

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: builder for '/nix/store/gimjnbf6gya35zsidan25zw6wqcyjmwj-python3.13-beautifulsoup4-4.13.4.drv' failed with exit code 1;                               
       last 25 log lines:                                                                                                                                    
       >                                                                                                                                                     
       >     def handle_charref(self, name: str) -> None:                                                                                                    
       >         """Handle a numeric character reference by converting it to the                                                                             
       >         corresponding Unicode character and treating it as textual                                                                                  
       >         data.                                                                                                                                       
       >                                                                                                                                                     
       >         :param name: Character number, possibly in hexadecimal.                                                                                     
       >         """                                                                                                                                         
       >         # TODO: This was originally a workaround for a bug in                                                                                       
       >         # HTMLParser. (http://bugs.python.org/issue13633) The bug has                                                                               
       >         # been fixed, but removing this code still makes some                                                                                       
       >         # Beautiful Soup tests fail. This needs investigation.                                                                                      
       >         if name.startswith("x"):                                                                                                                    
       >             real_name = int(name.lstrip("x"), 16)                                                                                                   
       >         elif name.startswith("X"):                                                                                                                  
       >             real_name = int(name.lstrip("X"), 16)                                                                                                   
       >         else:                                                                                                                                       
       > >           real_name = int(name)                                                                                                                   
       >                         ^^^^^^^^^                                                                                                                   
       > E           ValueError: invalid literal for int() with base 10: '2fontipt><!--</script><scriĐpt><┘┘┘┘╚┘┘┘┘┘┘┘pt><script<>!--</script6<\xa0<head/┬Xpe" charse/┬\x00ript>
<s\xa0■cript><!-pt><script<>!--</script6<\xa0<head/┬Xpe" charse/┬\x00ript><s\xa0■cript><!--</scri                                                                               
       >                                                                                                                                                                        
       > bs4/builder/_htmlparser.py:242: ValueError                                                                                                                             
       > =========================== short test summary info ============================                                                                                       
       > FAILED bs4/tests/test_fuzz.py::TestFuzz::test_deeply_nested_document_without_css[clusterfuzz-testcase-minimized-bs4_fuzzer-5984173902397440] - ValueError: invalid lite
ral for int() with base 10: '2fontipt><!--</script>...                                                                                                                          
       > ============ 1 failed, 842 passed, 7 skipped, 3 deselected in 1.02s ============
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.8, NixOS, 26.05 (Yarara), 26.05.20251204.835c620`
 - multi-user?: `yes`
 - sandbox: `yes`
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels' does not exist, ignoring
 - version: `nix-env (Lix, like Nix) 2.95.0-pre20251203-dev_3fcfedc
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/yuka/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/yuka/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/yuka/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/yuka/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/a98jk522y8c45cx8yfa20nblkljnhaph-lix-2.95.0-pre20251203-dev_3fcfedc/share`
 - nixpkgs: `/nix/store/1l10zj1vfgmgdk2y257wqlj03ydh52vm-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/beautifulsoup4/default.nix
+++ b/pkgs/development/python-modules/beautifulsoup4/default.nix
@@ -48,6 +48,15 @@ buildPythonPackage rec {
     hash = "sha256-YpKxxRhtNWu6Zp759/BRdXCZVlrZraXdYwvZ3l+n+4Y=";
   };
 
+  patches = [
+    # Fix tests with python 3.13.10 / 3.14.1
+    (fetchpatch {
+      url = "https://git.launchpad.net/beautifulsoup/patch/?id=55f655ffb7ef03bdd1df0f013743831fe54e3c7a";
+      excludes = [ "CHANGELOG" ];
+      hash = "sha256-DJl1pey0NdJH+SyBH9+y6gwUvQCmou0D9xcRAEV8OBw=";
+    })
+  ];
+
   build-system = [ hatchling ];
 
   nativeBuildInputs = [ sphinxHook ];

--- a/pkgs/development/python-modules/beautifulsoup4/default.nix
+++ b/pkgs/development/python-modules/beautifulsoup4/default.nix
@@ -35,7 +35,7 @@
 
 buildPythonPackage rec {
   pname = "beautifulsoup4";
-  version = "4.13.4";
+  version = "4.14.3";
   pyproject = true;
 
   outputs = [
@@ -45,17 +45,8 @@ buildPythonPackage rec {
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-27PE4c6uau/r2vJCMkcmDNBiQwpBDjjGbyuqUKhDcZU=";
+    hash = "sha256-YpKxxRhtNWu6Zp759/BRdXCZVlrZraXdYwvZ3l+n+4Y=";
   };
-
-  patches = [
-    # backport test fix for behavior changes in libxml 2.14.3
-    (fetchpatch {
-      url = "https://git.launchpad.net/beautifulsoup/patch/?id=53d328406ec8c37c0edbd00ace3782be63e2e7e5";
-      excludes = [ "CHANGELOG" ];
-      hash = "sha256-RtavbpnfT6x0A8L3tAvCXwKUpty1ASPGJKdks7evBr8=";
-    })
-  ];
 
   build-system = [ hatchling ];
 

